app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/ActivityAnalysis.java

```diff
@@ -65,12 +81,56 @@ ActivityAmounts calculateActivityAmounts(List<? extends ActivitySample> samples)
                     previousAmount.addSeconds(sharedTimeDifference);
                     amount.addSeconds(sharedTimeDifference);
                 }
+
+                // add time
+                if (steps > 0 && sample.getKind() == ActivityKind.TYPE_ACTIVITY) {
+                    if (steps > maxSpeed) {
+                        maxSpeed = steps;
+                    }
+
+                    if (!stats.containsKey(steps)) {
+                        //System.out.println("Adding: " + steps);
+                        stats.put(steps, (long) 60);
```

Comment:
In fact, this action adds a new step speed into stats HashMap. It assumes an initial length of 60s.